### PR TITLE
MAHOUT-1750 - For Review.  FlinkOpABt too many operations to implement?

### PR DESCRIPTION
**Pull request by andrewpalumbo**
When (partially) Implementing `FlinkOpABt`  so far, It seems that there may be too many Flink map/reduce/group/etc operations to finish using this method. **NOTE**: this is unfinished and is not numerically correct.

Currently, when testing am getting kyro stack overflow exceptions, which as I understand are often caused by a long string of operations: 
```
Job execution failed.
org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
	at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply$mcV$sp(JobManager.scala:716)
	at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply(JobManager.scala:662)
	at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply(JobManager.scala:662)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:401)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.StackOverflowError
	at com.esotericsoftware.kryo.util.DefaultClassResolver.writeClass(DefaultClassResolver.java:74)
	at com.esotericsoftware.kryo.Kryo.writeClass(Kryo.java:488)
	at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:57)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:495)
	at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:523)
{...}
```

Any comments are appreciated.

### CODE REVIEW

**Comment by rmetzger** [comment]
How can I reproduce the issue?

File: flink/src/main/scala/org/apache/mahout/flinkbindings/blas/FlinkOpABt.scala [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+      implicit val typeInformationJ = createTypeInformation[(Int, ((Array[K1], Matrix),(Int, (Array[K2], Matrix))))]
+      implicit val typeInformationJprod = createTypeInformation[(Int, T)]
+
+
+      // Perform the inner join.
+      val joined = blocksAKeyed.join(blocksBKeyed).where(0).equalTo(0)
+
+        // Apply product function which should produce smaller products. Hopefully, this streams blockB's in
+      val mapped = joined.rebalance().map{tuple => tuple._1._1 ->
```
Comment (smarthi) [inline flink/src/main/scala/org/apache/mahout/flinkbindings/blas/FlinkOpABt.scala]: u think rebalance() here is actually eliminating empty partitions ?

Comment (andrewpalumbo) [inline flink/src/main/scala/org/apache/mahout/flinkbindings/blas/FlinkOpABt.scala]: @smarthi  Not sure, I was testing a theory with that.  I should take that out.

**Comment by andrewpalumbo** [comment]
@rmetzger To reproduce, you'd have to pull this branch, and set a `$MAHOUT_HOME` env vaiabler to point at the base directory.  Then from  `$MAHOUT_HOME` run  `mvn clean install -DskipTests && cd flink && mvn test -Dsuites="RLikeDrmOpsSuite"`.

Or you could run the `RLikeDrmOpsSuite` suite from IntelliJ, but make sure that you set the `MAHOUT_HOME` env variable in the "Run/Debug Configurations"  for the test.

Thanks!

**Comment by andrewpalumbo** [comment]
or rather `mvn clean install -DskipTests && cd flink && mvn test -Dsuites="*RLikeDrmOpsSuite"`

**Comment by rmetzger** [comment]
I think the problem is the `TransposedMatrixView` class.

See :
recursive-mahout

And this code example

```scala
def main(args: Array[String]) {

    val ser = new KryoSerializer[TransposedMatrixView](classOf[TransposedMatrixView], new ExecutionConfig())
    val matrix = new SparseMatrix(15, 15)
    val inst = new TransposedMatrixView(matrix)
    ser.serialize(inst, new DataOutputSerializer(64))
  }
```

which will reproduce the error.

**Comment by andrewpalumbo** [comment]
Thank you very much for looking into this, Robert.  It is much appreciated.  I am having trouble understanding what the issue is with this class.  So you think that it is the Enumeration being returned by the `getStructure()` method in `TransposedMatrixView` that is the problem?

**Pull request closed without merging**